ui/src/app/utils/dates.tsx

```diff
@@ -31,6 +34,11 @@ export function displayDateWithoutHours(time: number): string {
   });
 }
 
+// If the time passed is null, return the nullDateStringRep else format it into date without hours
+export function formatDates(time: number, nullDateStringRep: string): string {
```

Comment:
this is a single date, so I'd prefer a singular name.

what do you think about adding a `?` to the parameter name, as a hint to the caller that null/undefined is ok?  (same Q for displayDateWithoutHours)